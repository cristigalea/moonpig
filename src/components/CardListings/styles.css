.card {
    align-items: flex-end;
    display: flex;
    position: relative;
    justify-content: center;
    float: left;
    margin: 10px;
    min-width: 270px;
    min-height: 190px;
    cursor: pointer;
}

.cardGrid {
    display: flex;
    flex-wrap: wrap;
    margin-left: 40px;
    margin-right: 40px;
    justify-content: space-around;
}

@media screen and (max-width: 640px) {
    .cardGrid {
        margin: 0;
    }
}

@media screen and (max-width: 500px) {
    .card {
        align-items: center;
        min-width: 320px;
    }
}