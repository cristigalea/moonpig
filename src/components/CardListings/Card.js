import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import LoadedImage from '../shared/LoadedImage';
import './styles.css';

const Card = ({ url, id }) => (
    <Link to={`/cards/${id}`} className="card">
        <LoadedImage url={url} />
    </Link>
);

Card.propTypes = {
    url: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
};

export default Card;