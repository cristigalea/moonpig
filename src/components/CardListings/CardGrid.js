import React from 'react';
import PropTypes from 'prop-types';

import Card from './Card';

const CardGrid = ({ items }) => {
    const cards = items.map((item, index) => <Card key={index} url={item.imageUrl} id={item.id} />);
    return (
        <div className="cardGrid">
            {cards}
        </div>
    );
};

CardGrid.propTypes = {
    items: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CardGrid;