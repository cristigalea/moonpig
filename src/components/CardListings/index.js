import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { getCardsList } from '../../actions/cardsActions';
import CardGrid from './CardGrid';
import './styles.css';

class CardListings extends Component {

    componentDidMount() {
        this.props.getCardsList();
    }

    render = () => <CardGrid items={this.props.cards} />;
};

export default withRouter(connect(
    state => ({
        cards: state.cards.list
    }),
    { getCardsList }
)(CardListings));