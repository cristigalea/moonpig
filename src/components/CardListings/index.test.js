import React from 'react';
import {shallow} from 'enzyme';

import CardListings from '.';

describe('<CardListings />', () => {

    const store = {
        cards: {
            list: [
                {
                    id: '1',
                    title: 'Test',
                    imageUrl: 'https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg'
                }
            ]
        }
    };

    it('should match the snapshot', () => {
        const component = shallow(<CardListings />, {store});
        expect(component).toMatchSnapshot();
    });
});