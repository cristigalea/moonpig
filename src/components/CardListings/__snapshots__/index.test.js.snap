// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardListings /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Connect(CardListings)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "store": Object {
      "cards": Object {
        "list": Array [
          Object {
            "id": "1",
            "imageUrl": "https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg",
            "title": "Test",
          },
        ],
      },
    },
  },
}
`;
